src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6916,6 +6916,11 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
                                 ReportNullabilityMismatchInAssignment(foreachSyntax.Type, sourceType, destinationType);
                             }
                         }
+                        else if (node.Syntax is ForEachStatementSyntax { Type: { IsVar: true } })
+                        {
+                            result = sourceState;
```

Comment:
Is this redundant with the assignment on 6909?

Reply:
Yes.

---
In reply to: [345396562](https://github.com/dotnet/roslyn/pull/39769#discussion_r345396562) [](ancestors = 345396562)